app/Models/Node.php

```diff
@@ -268,6 +270,11 @@ public function databaseHosts(): BelongsToMany
         return $this->belongsToMany(DatabaseHost::class);
     }
 
+    public function roles(): HasManyThrough
+    {
+        return $this->hasManyThrough(Role::class, RoleNode::class, 'node_id', 'id', 'id', 'role_id');
```

Comment:
Since we're using the expected column names, can we shorten this?

```suggestion
        return $this->hasManyThrough(Role::class, RoleNode::class);
```